[DOCS-5940] Add Container Images configuration info (#20865)
DataDog/documentation: content/en/infrastructure/containers/container_images.md

  tag: "Documentation"
  text: "Cloud Security Management Vulnerabilities"
- link: "/security/cloud_security_management/setup/csm_pro/?tab=aws#configure-the-agent-for-containers"
  tag: "Documentation"
  text: "Setting up container image vulnerabilities"
- link: "/security/vulnerabilities/troubleshooting"
  tag: "Documentation"
  text: "Troubleshooting Cloud Security Management Vulnerabilities"
---

## Overview

The [Container Images View][1] in Datadog provides key insights into every image used in your environment, helping you assess their deployment footprint, as well as detect and remediate security and performance issues that can affect multiple containers. You can view container image details alongside the rest of your container data, allowing you to troubleshoot image issues affecting infrastructure health. Additionally, you can view vulnerabilities found in your container images from [Cloud Security Management][2] (CSM) to help you streamline your security efforts.

{{< img src="security/vulnerabilities/container_images.png" alt="The Container Images view highlighting vulnerabilities and container column sort feature" width="100%">}}

## Configure Container Images View

> Images on the container images view are collected from several different sources (Live Containers, Image Collection, and Amazon ECR). The following instructions describe how to enable images from each of these sources.

Reviewer: Is it Container Image view or container images view or Container Images View?
```suggestion
Images on the container images view are collected from several different sources (Live Containers, Image Collection, and Amazon ECR). The following instructions describe how to enable images from each of these sources.
```

Reviewer 2: It is component of a product (Container Monitoring) -> so should be lower case, good call out though, let me go through the doc and make sure I have it consistent.

### Live Containers

> To enable live container collection, see the [containers][3] documentation for information on enabling the Process Agent, and excluding and including containers.

Reviewer: ```suggestion
To enable live container collection, see the [containers][3] documentation. It provides information on enabling the Process Agent, and excluding and including containers.
```

### Image collection

Datadog collects container image metadata to provide enhanced debugging context for related containers and [Cloud Security Management][8] (CSM) vulnerabilities.

#### Configure the Agent

> The following instructions enable the container image metadata collection and [Software Bill of Materials][5] (SBOM) collection in the Datadog Agent for CSM Vulnerabilities. This allows you to scan the libraries in container images to detect vulnerabilities. Vulnerabilities are evaluated and and scanned against your containers every hour.

Reviewer: ```suggestion
The following instructions enable the container image metadata collection and [Software Bill of Materials][5] (SBOM) collection in the Datadog Agent for CSM Vulnerabilities. This allows you to scan the libraries in container images to detect vulnerabilities. Vulnerabilities are evaluated and scanned against your containers every hour.
```

> **Note**: CSM Vulnerabilities is not available for AWS Fargate or Windows environments.

Reviewer: ```suggestion
**Note**: The CSM Vulnerabilities feature is not available for AWS Fargate or Windows environments.
```
or
```suggestion
**Note**: CSM Vulnerabilities are not available for AWS Fargate or Windows environments.
```

{{< img src="security/vulnerabilities/container_images.png" alt="The Container Images view highlighting vulnerabilities and container column sort feature" width="100%">}}

## Configure Container Images View

Images on the container images view are collected from several different sources (Live Containers, Image Collection, and Amazon ECR). The following instructions describe how to enable images from each of these sources.

### Live Containers

To enable live container collection, see the [containers][3] documentation for information on enabling the Process Agent, and excluding and including containers.

### Image collection

Datadog collects container image metadata to provide enhanced debugging context for related containers and [Cloud Security Management][8] (CSM) vulnerabilities.

#### Configure the Agent

> The following instructions enables the container image metadata collection and [Software Bill of Materials][5] (SBOM) collection in the Datadog Agent for CSM Vulnerabilities. This allows you to scan the libraries in container images to detect vulnerabilities. Vulnerabilities are evaluated and and scanned against your containers every hour.

Reviewer 3: Should this be "instructions enable"?
```suggestion
The following instructions enable the container image metadata collection and [Software Bill of Materials][5] (SBOM) collection in the Datadog Agent for CSM Vulnerabilities. This allows you to scan the libraries in container images to detect vulnerabilities. Vulnerabilities are evaluated and and scanned against your containers every hour.
```

---
title: Container Images
kind: documentation
further_reading:
- link: "/security/vulnerabilities"
  tag: "Documentation"
  text: "Cloud Security Management Vulnerabilities"
- link: "/security/vulnerabilities/troubleshooting"
  tag: "Documentation"
  text: "Troubleshooting Cloud Security Management Vulnerabilities"
- link: "/security/cloud_security_management/setup/csm_pro?tab=aws#configure-the-agent-for-hosts-and-containers"
  tag: "Documentation"
  text: "Setting up CSM container vulnerabilities"
- link: "/security/cloud_security_management/setup/csm_enterprise?tab=aws#configure-the-agent-for-vulnerabilities"
  tag: "Documentation"
  text: "Setting up CSM host vulnerabilities"
- link: "https://www.datadoghq.com/blog/datadog-container-image-view/"
  tag: "Blog"

>   text: "Enhance your troubleshooting workflow with Container Images in Datadog Container Monitoring"

Reviewer 4: Should we put this as first external link since it's closely related to the new page?

---

## Overview

> The [Container Images][1] view in Datadog provides key insights into every image used in your environment, helping you detect and remediate security and performance issues that can affect multiple containers. You can view container image details alongside the rest of your container data, allowing you to troubleshoot image issues affecting infrastructure health. Additionally, you can view vulnerabilities found in your container images from [Cloud Security Management][2] (CSM) to help you streamline your security efforts.

Reviewer 4: It wold be interesting to mention the ability to quickly assess deployment footprint using the new page.
Maye something like
```
The [Container Images][1] view in Datadog provides key insights into every image used in your environment, helping you assess their deployment footprint, detect and remediate security and performance issues that can affect multiple containers.
```

> {{< img src="security/vulnerabilities/container_images.png" alt="The Container Images tab highlighting vulnerabilities and container column sort feature" width="100%">}}

Reviewer 4: ```suggestion
{{< img src="security/vulnerabilities/container_images.png" alt="The Container Images view highlighting vulnerabilities and container column sort feature" width="100%">}}
```

## Configure container images view

> 

Reviewer 4: Maybe it's worth mentioning that container images are collected from different sources and then we can introduce each one of them with specific instruction on how they should be enabled.

### Live containers

> To enable live container collection, see the [containers][3] documentation for information on enabling the Process Agent, and excluding and including containers.

Reviewer 4: Should we capitalise Live Container since it's a product?

              }
            ]
        }
    ]
  ...
}
```

If the Agent fails to extract the SBOM from the container image, increase the Agent memory in the container definition:

```yaml
{
    "containerDefinitions": [
        {
            "name": "datadog-agent",
            "memory": 256,
            ...
        }
     ]
    ...
}
```
[1]: https://docs.datadoghq.com/containers/amazon_ecs/?tab=awscli#setup

{{% /tab %}}

{{% tab "Hosts" %}}

Add the following to your `datadog.yaml` configuration file:

```yaml
sbom:
  enabled: true
  container_image:
    enabled: true
container_image:
  enabled: true
```

[1]: /containers/amazon_ecs/?tab=awscli#setup
{{% /tab %}}
{{< /tabs >}}

### Container registries

#### Amazon Elastic Container Registry (Amazon ECR)

Set up the [AWS integration][4] to begin crawling Container Image metadata from Amazon ECR.

## Container image tagging

Containers are [tagged][6] with all existing host-level tags, as well as with metadata associated with individual containers. All containers are tagged by `image_name`, including integrations with popular orchestrators, such as ECS and Kubernetes, which provide further container-level tags. 

> If you have a configuration for [Unified Service Tagging][7] in place, Datadog automatically picks up `env`, `service`, and `version` tags. 

Reviewer 4: Container Image tagging is not related to Live Container tags. As of today, if users want to tag their Images with arbitrary tags, the only supported way is by using the "Extract Labels as Tags" on the agent. These tags will then be picked by the Container Image check.
https://docs.datadoghq.com/containers/docker/tag/?tab=containerizedagent#extract-labels-as-tags

Reviewer 4: Would it be possible to add a `team` tag on the screenshot you've shared?

Eventually, we'd like to add a link to this doc from the Container Images view when the Team Lens is enabled so having an example with team will be useful

![Container Images Datadog 2023-12-20 at 3 41 37 PM](https://github.com/DataDog/documentation/assets/45740044/c7efce9c-40a4-4d1d-aed6-cd745f210722)

Reviewer 2: Hi @moisesbotarro for the screenshot, are you wanting just a screen shot of the Container Images view with a team filter (such as below)? Or were you wanting a screenshot of the details side panel of a specific container where it has an arbitrary `team` container tag? I'm not finding an example of the latter, i.e. if I filter by `team:mobile` and then select one of those images, the `team` tag doesn't seem to carry over. Perhaps because "extract labels as tags" isn't configured. Let me know if this is what you were looking for if you have an example of one ? Thanks!

<img width="1839" alt="Container Images Datadog 2023-12-20 at 1 35 49 PM" src="https://github.com/DataDog/documentation/assets/26307719/f2f2931d-3018-4dfe-82ac-2697991be0f9">

Reviewer 4: HI @aliciascott, what I had in mind is to show arbitrary tags in the tag list on the details side panel like the screenshot you had before.

> I'm not finding an example of the latter, i.e. if I filter by team:mobile and then select one of those images, the team tag doesn't seem to carry over.

This is expected because this `team` tag is coming from the containers running the image and it's not a tag of the image itself.

> Perhaps because "extract labels as tags" isn't configured.

You are right. I was looking for other examples and I don't think that "extract labels as tags" is set up yet.

What do you think of only mentioning that users can enrich their container images with arbitrary tags and link to the "Extract Labels as Tags" doc, without any screenshot?

Reviewer 2: Sounds good to me, per our slack convo, we can work on setting this up later and iterate on the docs.
